.github/workflows/nightly-release.yml

```diff
@@ -0,0 +1,100 @@
+name: Nightly Release
+
+on:
+  # TODO: Remove once this workflow is in the main branch
+  push:
+    branches: [ add_nightly_release ]
+  workflow_dispatch:
+
+jobs:
+  rtl-repo-sync:
+    name: RTL Repo Sync
+    uses: ./.github/workflows/rtl-repo-sync.yml
+
+  find-latest-release:
+    name: Find Latest Release
+    needs: rtl-repo-sync
+    runs-on: ubuntu-22.04
+    outputs:
+      create_release: ${{ steps.find.outputs.create_release }}
+      new_release_tag: ${{ steps.find.outputs.new_release_tag }}
+      release_ref: ${{ steps.find.outputs.release_ref }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+          ref: 'main'
+          fetch-depth: 0
+
+      - name: Find latest release
+        id: find
+        run: |
+          git config --global user.name "GitHub CI"
+          git config --global user.email "username@users.noreply.github.com"
+          TAG_BASE=release_v
+          INDEX=0
+          MOST_RECENT_RELEASE=None
+          while git tag | grep ${TAG_BASE}${INDEX}
+          do
+              MOST_RECENT_RELEASE=$(git tag | grep ${TAG_BASE}${INDEX} | head -1)
+              ((INDEX+=1))
+          done
+          if [[ "$MOST_RECENT_RELEASE" == "None" ]]; then
+              echo "create_release=true" >> $GITHUB_OUTPUT
+          else
+              COMMITS_AFTER_LAST_RELEASE=$(git rev-list --count $MOST_RECENT_RELEASE..HEAD)
+              if [[ $COMMITS_AFTER_LAST_RELEASE -gt 0 ]]; then
+                  echo "create_release=true" >> $GITHUB_OUTPUT
+              else
+                  echo "create_release=false" >> $GITHUB_OUTPUT
+              fi
+          fi
+          echo "new_release_tag=${TAG_BASE}${INDEX}" >> $GITHUB_OUTPUT
+          echo "release_ref=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+          echo "New release tag: ${TAG_BASE}${INDEX}"
+
+  verilator-full-suite:
+    name: Nightly Verilator Suite
+    needs: find-latest-release
+    if: needs.find-latest-release.outputs.create_release
+    uses: ./.github/workflows/nightly-verilator.yml
+    with:
+      ref: ${{ needs.find-latest-release.outputs.release_ref }}
+
+  create-release:
+    name: Create New Release
+    needs: [find-latest-release, verilator-full-suite]
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Get current date
+        id: date
+        run: echo "date=$(date +'%Y%m%d')" >> $GITHUB_OUTPUT
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+          ref: ${{ needs.find-latest-release.outputs.release_ref }}
+
+      - name: Generate release zip
+        run: |
+          mkdir release_${{ steps.date.outputs.date }}
+          cargo run --manifest-path=builder/Cargo.toml --bin image -- --rom caliptra_rom.bin --fw image_bundle.bin
```

Comment:
We should have a single script that we call here that generates the release (to let us strip out large files that are not needed, etc)